crd/multitenantpodnetworkconfig/client.go

```diff
@@ -0,0 +1,161 @@
+}
+
+func NewInstaller(c *rest.Config) (*Installer, error) {
+	cli, err := crd.NewCRDClientFromConfig(c)
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to init crd client")
+	}
+	return &Installer{
+		cli: cli,
+	}, nil
+}
+
+func (i *Installer) create(ctx context.Context, res *v1.CustomResourceDefinition) (*v1.CustomResourceDefinition, error) {
+	res, err := i.cli.Create(ctx, res, metav1.CreateOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create mpnc crd")
+	}
+	return res, nil
+}
+
+// Installs the embedded MultitenantPodNetworkConfig CRD definition in the cluster.
+func (i *Installer) Install(ctx context.Context) (*v1.CustomResourceDefinition, error) {
+	mpnc, err := GetMultitenantPodNetworkConfigs()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get embedded mpnc crd")
+	}
+	return i.create(ctx, mpnc)
+}
+
+// InstallOrUpdate installs the embedded MultitenantPodNetworkConfig CRD definition in the cluster or updates it if present.
+func (i *Installer) InstallOrUpdate(ctx context.Context) (*v1.CustomResourceDefinition, error) {
+	mpnc, err := GetMultitenantPodNetworkConfigs()
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to get embedded mpnc crd")
+	}
+	current, err := i.create(ctx, mpnc)
+	if !apierrors.IsAlreadyExists(err) {
+		return current, err
+	}
+	if current == nil {
+		current, err = i.cli.Get(ctx, mpnc.Name, metav1.GetOptions{})
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to get existing mpnc crd")
+		}
+	}
+	if !reflect.DeepEqual(mpnc.Spec.Versions, current.Spec.Versions) {
+		mpnc.SetResourceVersion(current.GetResourceVersion())
+		previous := *current
+		current, err = i.cli.Update(ctx, mpnc, metav1.UpdateOptions{})
+		if err != nil {
+			return &previous, errors.Wrap(err, "failed to update existing mpnc crd")
+		}
+	}
+	return current, nil
+}
+
+// Client provides methods to interact with instances of the MultitenantPodNetworkConfig custom resource.
+type Client struct {
+	cli client.Client
+}
+
+// NewClient creates a new MultitenantPodNetworkConfig client around the passed ctrlcli.Client.
+func NewClient(cli client.Client) *Client {
+	return &Client{
+		cli: cli,
+	}
+}
+
+// Get returns the MultitenantPodNetworkConfig identified by the NamespacedName.
+func (c *Client) Get(ctx context.Context, key types.NamespacedName) (*v1alpha1.MultitenantPodNetworkConfig, error) {
+	multitenantPodNetworkConfig := &v1alpha1.MultitenantPodNetworkConfig{}
+	err := c.cli.Get(ctx, key, multitenantPodNetworkConfig)
+	return multitenantPodNetworkConfig, errors.Wrapf(err, "failed to get mpnc %v", key)
+}
+
+// PatchSpec performs a server-side patch of the passed MultitenantPodNetworkConfigSpec to the MultitenantPodNetworkConfig specified by the NamespacedName.
+func (c *Client) PatchSpec(ctx context.Context, key types.NamespacedName, spec *v1alpha1.MultitenantPodNetworkConfigSpec, fieldManager string) (*v1alpha1.MultitenantPodNetworkConfig, error) {
```

Comment:
is this necessary? MTPNC will get created with known values for spec and I don't expect the spec to need to get updated

Reply:
removed